pkg/interfaces/interfaces.go

```diff
@@ -78,3 +78,39 @@ func DetachXDPFromAllInterfaces() error {
 	}
 	return nil
 }
+
+// GetInterfaceIndices return one or more interface index based on the interface type
+// Note: for bond interfaces we attach XDP to the bond interface but the xdp packets
+// will be using bond member's interface_indices not the bond interface_index.
+func GetInterfaceIndices(interfaceName string) ([]uint32, error) {
+	var membersList []uint32
+
+	link, err := netlink.LinkByName(interfaceName)
+	if err != nil {
+		return nil, err
+	}
+
+	if link.Type() != "bond" {
```

Comment:
Are we sure the change in behavior only applies to bond interfaces and not any other types? And that this is the best way to identify bond interfaces?

Also, will this change mean that now the latest version of infw doesn't work correctly with bonds on RHEL8?

Should we add rules for both the member ifIDs and the master ifID?

Reply:
the latest won't work with rhel8 for other things as well like enabling jumpo supporyt/apis, we can't enable XDP on members and bond we will get error already exists